ui/app/routes/datasets/$dataset_name/route.tsx

```diff
@@ -60,6 +60,8 @@ export default function DatasetDetailPage({
         description: `Added ${rowsAdded} rows to the dataset. Skipped ${rowsSkipped} duplicate rows.`,
       });
     }
+    // TODO: Fix and stop ignoring lint rule
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
also for my learning: what does this mean?

Reply:
It just means we should update the dependency array to include all dependencies, but I'm not sure doing that in some cases will not have unintended consequences. It's probably safe in some cases but it's also possible we can remove some of these effects altogether, so I want to hit them each one at a time to be sure.